### PR TITLE
fixed issue with the read filter in case of nested AND and OR for AVRO readformat

### PR DESCRIPTION
**Pull request by himanshukohli09**

### CODE REVIEW

File: connector/src/test/java/com/google/cloud/spark/bigquery/SparkFilterUtilsTest.java [hunk 1]
```diff
@@ -188,4 +188,93 @@ public void testTimestampFilters() throws ParseException {
...
+
+    Filter filter =
+        Or.apply(
+            And.apply(GreaterThanOrEqual.apply("c1", 500), LessThanOrEqual.apply("c2", 300)),
+            And.apply(LessThanOrEqual.apply("c1", 800), GreaterThanOrEqual.apply("c3", 230)));
+
+    checkFilters(
+        AVRO,
+        "",
+        "(((((`c1` >= 500) AND (`c2` <= 300))) OR (((`c1` <= 800) " + "AND (`c3` >= 230)))))",
```
Comment (davidrabinowitz) [inline connector/src/test/java/com/google/cloud/spark/bigquery/SparkFilterUtilsTest.java:236]: Please remove the string concatenation

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: connector/src/test/java/com/google/cloud/spark/bigquery/SparkFilterUtilsTest.java (hunk 1 above)
Comment (himanshukohli09) [reply connector/src/test/java/com/google/cloud/spark/bigquery/SparkFilterUtilsTest.java]: Done

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**